src/lib/types/events.ts

```diff
@@ -3,115 +3,222 @@ import { FeatureToggle, IStrategyConfig, ITag, IVariant } from './model';
+export const TAG_TYPE_IMPORT = 'tag-type-import' as const;
+export const DROP_TAG_TYPES = 'drop-tag-types' as const;
+export const ADDON_CONFIG_CREATED = 'addon-config-created' as const;
+export const ADDON_CONFIG_UPDATED = 'addon-config-updated' as const;
+export const ADDON_CONFIG_DELETED = 'addon-config-deleted' as const;
+export const DB_POOL_UPDATE = 'db-pool-update' as const;
+export const USER_CREATED = 'user-created' as const;
+export const USER_UPDATED = 'user-updated' as const;
+export const USER_DELETED = 'user-deleted' as const;
+export const DROP_ENVIRONMENTS = 'drop-environments' as const;
+export const ENVIRONMENT_IMPORT = 'environment-import' as const;
+export const SEGMENT_CREATED = 'segment-created' as const;
+export const SEGMENT_UPDATED = 'segment-updated' as const;
+export const SEGMENT_DELETED = 'segment-deleted' as const;
+export const GROUP_CREATED = 'group-created' as const;
+export const GROUP_UPDATED = 'group-updated' as const;
+export const SETTING_CREATED = 'setting-created' as const;
+export const SETTING_UPDATED = 'setting-updated' as const;
+export const SETTING_DELETED = 'setting-deleted' as const;
+
+export const CLIENT_METRICS = 'client-metrics' as const;
+export const CLIENT_REGISTER = 'client-register' as const;
+
+export const PAT_CREATED = 'pat-created' as const;
+export const PAT_DELETED = 'pat-deleted' as const;
+
+export const PUBLIC_SIGNUP_TOKEN_CREATED =
+    'public-signup-token-created' as const;
+export const PUBLIC_SIGNUP_TOKEN_USER_ADDED =
+    'public-signup-token-user-added' as const;
+export const PUBLIC_SIGNUP_TOKEN_TOKEN_UPDATED =
+    'public-signup-token-updated' as const;
+
+export const CHANGE_REQUEST_CREATED = 'change-request-created' as const;
+export const CHANGE_REQUEST_DISCARDED = 'change-request-discarded' as const;
+export const CHANGE_ADDED = 'change-added' as const;
+export const CHANGE_DISCARDED = 'change-discarded' as const;
+export const CHANGE_REQUEST_APPROVED = 'change-request-approved' as const;
+export const CHANGE_REQUEST_APPROVAL_ADDED =
+    'change-request-approval-added' as const;
+export const CHANGE_REQUEST_CANCELLED = 'change-request-cancelled' as const;
+export const CHANGE_REQUEST_SENT_TO_REVIEW =
+    'change-request-sent-to-review' as const;
+export const CHANGE_REQUEST_APPLIED = 'change-request-applied' as const;
+
+export const API_TOKEN_CREATED = 'api-token-created' as const;
+export const API_TOKEN_UPDATED = 'api-token-updated' as const;
+export const API_TOKEN_DELETED = 'api-token-deleted' as const;
+
+export const FEATURE_FAVORITED = 'feature-favorited' as const;
+export const FEATURE_UNFAVORITED = 'feature-unfavorited' as const;
+export const PROJECT_FAVORITED = 'project-favorited' as const;
+export const PROJECT_UNFAVORITED = 'project-unfavorited' as const;
+export const FEATURES_EXPORTED = 'features-exported' as const;
+export const FEATURES_IMPORTED = 'features-imported' as const;
+
+export const IEventTypes = [
```

Comment:
@chriswk by making it this strict, we can't build unleash-enterprise, as there are types in enterprise which are not defined here: https://github.com/ivarconr/unleash-enterprise/actions/runs/5473805611/jobs/9967754171 I think we should make it string again